app/src/main/java/com/ae/apps/tripmeter/fragments/TripsListFragment.java

```diff
@@ -103,10 +154,30 @@ public void onClick(View v) {
                 showAddTripDialog();
             }
         });
-
         return view;
     }
 
+
+    @Override
+    public void onRequestPermissionsResult(int requestCode, @NonNull String[] permissions, @NonNull int[] grantResults) {
+        switch (requestCode) {
+            case PERMISSIONS_REQUEST_READ_CONTACTS: {
+                if (grantResults.length > 0
+                        && grantResults[0] == PackageManager.PERMISSION_GRANTED) {
+                    //TODO: Reload this Fragment.Otherwise the FAB won't work
```

Comment:
Is this issue caused by onRequestPermissionsResult being Async. Could we just have the code to paint the fragment called here before requestPermissions was implemented?

onCreateView() -> return an empty view
onRequestPermissionsResult()
    PERMISSION_GRANTED -> inflate(R.layout.fragment_trips_list, container, false); and add to view// we could cache the inflator
    PERMISSION_DENIED -> view.add(new TextView()); // add to empty view already created